### PR TITLE
improve dev build times for everyone

### PR DESCRIPTION
**Pull request by tbro**
Adds dev profile to `Cargo.toml`. It improves build times considerably. I don't think it hurts anything. So probably ok if CI passes. Feel free to run some cargo commands to see how you like it.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -156,3 +156,12 @@ hotshot-testing ={ git = "https://www.github.com/EspressoSystems/HotShot.git", t
 hotshot-types = { git = "https://www.github.com/EspressoSystems/HotShot.git", tag = "0.5.75-patch1" }
 libp2p-networking = { git = "https://github.com/EspressoSystems//HotShot.git", tag = "0.5.75-patch1" }
 hotshot-example-types = { git = "https://www.github.com/EspressoSystems/HotShot.git", tag = "0.5.75-patch1" }
+
+[profile.dev]
+opt-level = 1
```
Comment (ss-es) [inline Cargo.toml:161]: Is there a reason not to set this (and the other one) to just `opt-level = 0`?

File: Cargo.toml [hunk 2]
```diff
@@ -156,3 +156,12 @@ hotshot-testing ={ git = "https://www.github.com/EspressoSystems/HotShot.git", t
...
+# Skip compiling the debug information in the first place.
+debug = 0
+# Skip linking debug information.
+strip = "debuginfo"
```
Comment (ss-es) [inline Cargo.toml:165]: is there a reason not to just do `strip = true`? (I don't actually know how significant this is)

Comment (tbro) [reply Cargo.toml:165]: I know that we don't want `debuginfo`. I'm not sure if we want `symbols` or not.

File: Cargo.toml (hunk 1 above)
Comment (tbro) [reply Cargo.toml]: I don't remember why I set it to 1.

f329bdf1e61d7729d06d48934265b7957b46edaa

**Review by sveitser — APPROVED** [review]

**Pull request merged**